Sources/Apollo/InterceptorRequestChain.swift

```diff
@@ -9,28 +9,26 @@ final public class InterceptorRequestChain: Cancellable, RequestChain {
   public enum ChainError: Error, LocalizedError {
     case invalidIndex(chain: RequestChain, index: Int)
     case noInterceptors
+    case unknownInterceptor(id: String)
 
     public var errorDescription: String? {
       switch self {
       case .noInterceptors:
         return "No interceptors were provided to this chain. This is a developer error."
       case .invalidIndex(_, let index):
         return "`proceedAsync` was called for index \(index), which is out of bounds of the receiver for this chain. Double-check the order of your interceptors."
+      case let .unknownInterceptor(id):
+        return "`proceedAsync` was called by unknown interceptor \(id)."
       }
     }
   }
 
-  private var interceptors: [ApolloInterceptorReentrantWrapper]
+  private var interceptors: [any ApolloInterceptor]
+  private var interceptorIndexes: [AnyHashable: Int] = [:]
```

Comment:
A new lookup table that gives us a faster way to find an interceptors index.

Reply:
Good call on using a hash map to improve performance here!

If the interceptor id is always a `String` can't this just be a `String: Int` dictionary?

Reply:
Why are `interceptors` and `interceptorIndexes` variables? Could they be `let` properties, since they are created during initialization?

Reply:
> If the interceptor id is always a String can't this just be a String: Int dictionary?

Yes, it can be. I thought I'd updated that when I moved away from an `associatedtype`. I'll update it.

> Why are interceptors and interceptorIndexes variables? Could they be let properties, since they are created during initialization?

`interceptors` and `callbackQueue` can become `let` but `interceptorIndexes` needs to be `var` because of the way we enumerate through `interceptors` to assign the `index` to the `id`.